### PR TITLE
Resolve dependencies adapter

### PR DESCRIPTION
**Pull request by Mitch391**

### CODE REVIEW

File: src/ObjectBridgeService/ResolveDependenciesAdapter.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use Pimcore\Model\DataObject\AbstractObject;
+
+class ResolveDependenciesAdapter
+{
+
+    public const CONCRETE_DATA_TYPE = 'object';
+    
+    /**
+     * Default functionality from ClassDefinition\Data\Object::resolveDependencies
+     * @param AbstractObject ...$abstractObjects
```
Comment (JeroenVanDerLaan) [inline src/ObjectBridgeService/ResolveDependenciesAdapter.php]: Can be `@return array[]`

File: src/ObjectBridgeService/ResolveDependenciesAdapter.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            $dependencies['object_' . $id] = $dependency;
+        }
+
+        return $dependencies;
+    }
+
+    /**
+     * @param $id
+     * @return array
+     */
```
Comment (JeroenVanDerLaan) [inline src/ObjectBridgeService/ResolveDependenciesAdapter.php]: Missing `$id` type hint

File: tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+use Pimcore\Model\DataObject\AbstractObject;
+use Youwe\ObjectBridge\ObjectBridgeService\ResolveDependenciesAdapter;
+
+class ResolveDependenciesAdapterTest extends TestCase
+{
+    /**
+     * @var ResolveDependenciesAdapter
+     */
+    private $adapter;
+
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php]: A little nitpicky, but if this is supposed to be 1 mock, why not pop the mock from the array?

File: tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public function testResolveDependenciesSingleInputReturnsCorrectKey(): void
+    {
+        $array = $this->adapter->resolveDependencies(...$this->oneMock);
+        $this->assertArrayHasKey("object_1", $array);
+    }
+
+    public function testResolveDependenciesSingleInputReturnsCorrectId(): void
+    {
+        $array = $this->adapter->resolveDependencies(...$this->oneMock);
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php]: Extract the second dimension element from the array before trying to access it. If `$array["object_id"]` does not exist, PHP might complain and possibly kill test suite execution.

I see that you tested for the correct key already. [But in principle, each test (aka test methods) should be able to run independently and in no particular order.](https://github.com/ghsukumar/SFDC_Best_Practices/wiki/F.I.R.S.T-Principles-of-Unit-Testing)

Comment (Mitch391) [reply tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php]: should i just add what i tested in the previous test to this one then?

Comment (JeroenVanDerLaan) [reply tests/unit/ObjectBridgeService/ResolveDependenciesAdapterTest.php]: Nha, I would just securely fetch the value that you want to test (the arrangement step of a test), and assert that the value is what you expect. Basically not giving php any room to complain.
Something like this..
```
$data = isset($array["object_1"]) ? $array["object_1"] : [];
$actualId = isset($data["id"]) ? $data["id"] : null;
$this->assertEqual($expectedId, $actualId);
```

**Pull request merged**